### ISSUE TITLE
Some Syntax Errors Inside `eval` Aborting Interpreter Instead of Raising `SyntaxError`

### ISSUE THREAD

**Issue by edgemaster** [resource-reasoning/JavaScriptVerification#1]
For example: test262-es6/test/language/statements/with/12.10.1-10-s.js

**Comment by edgemaster** [comment]
6 other tests also failing due to this. (And I guess more with other types of unsupported syntax also).

**Comment by edgemaster** [comment]
https://github.com/resource-reasoning/JavaScriptVerification/blob/master/src/pulp/syntax/pulp_Translate.ml#L1477
We should conditionally throw Error based on mode, something like
```ocaml
if strictMode then raise EarlySyntaxError else raise PulpNotImplemented end
```

**Comment by edgemaster** [comment]
Fixed in dfecdca
